### ISSUE TITLE
Why Is `--workspace-folder` Required or Default to $PWD?

### ISSUE THREAD

**Issue by metaskills** [devcontainers/cli#29]
I found it odd that this parameter was required when using `devcontainer up` and was wondering why and if that was intentional. Not sure if it is also required in other commands ( just learning here ) but did want to share it felt odd to do this.

```shell
$ devcontainer up --workspace-folder .
```

**Comment by hpe-ykoehler** [comment]
Even more confusing is the devcontainer build [path] which still require --workspace-folder, seems like a bug, either path or workspace-folder but not both please.

**Comment by alexec** [comment]
Having to type `--workspace-folder .` all the time is crazy annoying. Please default to $PWD.

**Comment by hpe-ykoehler** [comment]
> Having to type `--workspace-folder .` all the time is crazy annoying. Please default to $PWD.

And add support to specify using an ENV variable as well, DEVCONTAINER_WORKSPACE_FOLDER or whatever.

**Comment by MunsMan** [comment]
I've created a PR #425, with a basic check if a `.devcontainer` directory exists in your `cwd`. I would be pretty easy to add an ENV variable to it, by adding it to the `findWorkspaceFolder` Function.

@hpe-ykoehler Currently, I'm not convinced if an ENV variable is needed. Do you have a use case for that?

**Comment by hpe-ykoehler** [comment]
@MunsMan no, but convention is that config option can be controlled by cli argument, env or config file in that order of priority.  Agree env may appears less useful, but devcontainer should use a CLI framework that handles all those details (such as Cobra if written in Go) no?

**Comment by felipecrs** [comment]
I'd suggest not to add the environment variable support unless someone asks.

**Comment by MunsMan** [comment]
I added this function to my `.zshrc`, which does not solve the problem, but reduces the pain:

```bash
function dev() {
     devcontainer $1 --workspace-folder . ${@: 2}
 }
```

Now I can just call `dev up` or `dev exec ls`

Maybe it helps someone while the Issue exists.

**Comment by eric-anderton-at-rearc** [comment]
> I added this function to my `.zshrc`, which does not solve the problem, but reduces the pain:
> 
> ```shell
> function dev() {
>      devcontainer $1 --workspace-folder . ${@: 2}
>  }
> ```
> 
> Now I can just call `dev up` or `dev exec ls`
> 
> Maybe it helps someone while the Issue exists.

I did something similar that also takes an optional environment var.  The following will set `--workspace-folder` to  `$WORKSPACE_FOLDER` if it is set, defaulting to the present working directory (`$PWD`).

```bash
function dev {

workspace_folder=${WORKSPACE_FOLDER:-$PWD}

case $1 in
'up' | 'build' | 'run-user-commands' | 'read-configuration' | 'exec')
  # specify the workspace folder to commands that need it
  dev_cmd=$1
  shift 1
  devcontainer "$dev_cmd" --workspace-folder "$workspace_folder" "$@"
  ;;  
*)
  # directly run everything else
  devcontainer "$@"
esac
}
```

That said, please don't take my contribution of a robust BASH warpper as any indication that this isn't an issue.  Things like may seem small, but are problematic for the ergonomics of this CLI tool.

**Comment by shish** [comment]
> We should be able to pick this up soon

Polite ping :)

I see three people have attempted to do this in a smart way (recursing up parent directories looking for either `.devcontainer` or `.devcontainer.json`) and all ended up either abandoned or in review hell; perhaps it's worth doing the dumb and simple thing of _just_ "set the default to '.'" in order to satisfy the most common use-case, and the fancy recursion can be added as a backwards compatible enhancement later?

**Comment by agardnerIT** [comment]
We've just hit this too. I too vote that it should default to `pwd`

**Comment by chrmarti** [comment]
Now available with https://github.com/devcontainers/cli/issues/1069. Thanks.
